commons/src/main/java/fr/free/nrw/commons/category/PrefixUpdater.java

```diff
@@ -0,0 +1,77 @@
+package fr.free.nrw.commons.category;
+
+import android.os.AsyncTask;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+
+import org.mediawiki.api.ApiResult;
+import org.mediawiki.api.MWApi;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import fr.free.nrw.commons.CommonsApplication;
+
+public class PrefixUpdater extends AsyncTask<Void, Void, ArrayList<String>> {
```

Comment:
Hm, I could try to change it to a List but that would involve changing a lot of the preexisting code (i.e. the code that existed before I took on this project). For instance catFragment.recentCatQuery(); returns an ArrayList, catFragment.categoriesCache.get(filter); also returns an ArrayList, etc. Do you think the advantages of List are worth modifying the old code for?